### PR TITLE
replace failure with thiserror

### LINKED ISSUE #19: Possibly migrate from failure to thiserror
Since failure is deprecated and recommends thiserror & anyhow, it'd be nice to migrate away from it in a future release. 

Failure use in libraries we've found a bit tricky to integrate with our own non-failure error handling at least.

### PR DESCRIPTION
**Pull request by darkskygit**
fix #19

### CODE REVIEW

**Comment by ohadravid** [comment]
Nice! I would have liked to improve the API by using only thiserror, but this is a nice improvement already 🙌🏻

Seems like there is a non related tests error in
https://github.com/ohadravid/wmi-rs/blob/master/src/query.rs#L638
Would you mind changing this to 'assert(true)'? After the tests pass I'll merge this.

Thanks for this PR 🚀

File: Cargo.toml [hunk 1]
```diff
@@ -21,19 +21,20 @@ default-target = "x86_64-pc-windows-msvc"
...
-winapi = { version = "0.3.7", features = ["objbase", "wbemcli", "objidlbase", "oaidl", "oleauto", "errhandlingapi"] }
-failure = "0.1.5"
+winapi = { version = "0.3.9", features = ["objbase", "wbemcli", "objidlbase", "oaidl", "oleauto", "errhandlingapi"] }
 log = "0.4"
 widestring = "0.4"
 serde = { version = "1.0", features = ["derive"] }
 chrono = { version = "0.4", features = ["serde"] }
-lazy_static = { version = "1.2.0", optional = true }
+lazy_static = { version = "1.4.0", optional = true }
+anyhow = "1.0.31"
```
Comment (ohadravid) [inline Cargo.toml:30]: We should probably pin to `1^`

File: src/query.rs [hunk 2]
```diff
@@ -636,7 +634,7 @@ mod tests {
             for res in results {
                 match res.get("Caption") {
                     Some(Variant::String(s)) => assert!(s != ""),
-                    _ => assert!(false),
+                    _ => assert!(true),
```
Comment (ohadravid) [inline src/query.rs:637]: I'm sorry I meant this:

```suggestion
                    Some(Variant::String(s)) => assert!(true),
                    _ => assert!(false),
```

**Comment by darkskygit** [comment]
i made some modifications to the error handling to remove anyhow, please wait

**Comment by darkskygit** [comment]
the code now work :)

File: Cargo.toml [hunk 3]
```diff
@@ -21,19 +21,19 @@ default-target = "x86_64-pc-windows-msvc"
...
-winapi = { version = "0.3.7", features = ["objbase", "wbemcli", "objidlbase", "oaidl", "oleauto", "errhandlingapi"] }
-failure = "0.1.5"
+winapi = { version = "0.3.9", features = ["objbase", "wbemcli", "objidlbase", "oaidl", "oleauto", "errhandlingapi"] }
 log = "0.4"
 widestring = "0.4"
 serde = { version = "1.0", features = ["derive"] }
 chrono = { version = "0.4", features = ["serde"] }
-lazy_static = { version = "1.2.0", optional = true }
+lazy_static = { version = "1.4.0", optional = true }
+thiserror = "1.0.20"
```
Comment (ohadravid) [inline Cargo.toml:30]: Pin to "^1"

File: src/query.rs [hunk 4]
```diff
@@ -498,20 +499,12 @@ mod tests {
...
-                        match wmi_err {
-                            WMIError::HResultError { hres } => {
-                                assert_eq!(*hres, WBEM_E_INVALID_QUERY as HRESULT);
-                            }
-                            _ => assert!(false),
-                        }
-                    } else {
-                        assert!(false);
+                Err(wmi_err) => match wmi_err {
+                    WMIError::HResultError { hres } => {
```
Comment (ohadravid) [inline src/query.rs:503]: 🎉 this is so much nicer!

**Pull request merged**